### ISSUE TITLE
Forms 8.7.6 - Checkbox always saved as "False" on submit

### ISSUE THREAD

**Issue by JeroenRoos** [umbraco/Umbraco.Forms.Issues#615]
Umbraco Forms: 8.7.6
Umbraco CMS: 8.12.2

One of our clients is using the Checkbox field type in their forms, and the value is always saved as "False".  Even if the checkbox is checked. This happens in both the Submission in Forms, and the email with the form values that is sent through a Workflow.

Here is the HTML we use for the Checkbox, which is a slightly edited version of the standard Checkbox view:

    <div class="custom-control custom-checkbox">
    <input type="hidden" name="@Model.Name" value="false" /> <input type="checkbox"
           name="@Model.Name"
           id="@Model.Id"
           value="true"
           data-umb="@Model.Id"
           class="custom-control-input"
           @if (Model.Mandatory) { <text> data-val="true" data-val-requiredcb="@Model.RequiredErrorMessage" </text> }
           @if (Model.ContainsValue(true) || Model.ContainsValue("true") || Model.ContainsValue("on")) { <text> checked="checked" </text> } />

    <label class="custom-control-label" for="@Model.Id">
        @Model.Caption
    </label> </div>

**Comment by AndyButland** [comment]
Have just tested this - am not finding any problems with the out of the box template, but when I swap it for your custom one, I see the same as you, with the value recorded as "False".

The only difference I can see between your mark-up and what we ship with the package, is that in yours the hidden field is before the check-box, and in the standard template it's after.  Sure enough, switching them round in your mark-up looks to solve the problem - and checked values are correctly recorded.

It's perhaps not obvious why that should matter, but digging into the code a bit we have this logic when saving the value:

```
            var values = new List<object>();
            var enumerable = postedValues.ToArray();

            if (enumerable.Any() && enumerable.First().ToString() == "true")
            {
                values.Add(true);
            }
            else
            {
                values.Add(false);
            }

            return values;
```

Hence you can see there that if we have a posted value of "false, true" rather than "true, false", we'll record the value as false rather than true.

So in short, please just try switching around the mark-up so the hidden field is after the checkbox field, and that should sort out the issue.
